### PR TITLE
[Maintained] Upgrade to JDA 5 (currently: .20)

### LINKED ISSUE #46: Issue with newer Versions of JDA
Due to several classes in JDA being moved around, implementations/class hierarchy changed and/or interactions having been rewritten in the newer versions of the JDA, the pagination util does not fully work for the newest version.

I wanted to open this issue merely to point them out in case you wanted to fix them on your end. When the relevant adjusments are made, the library works perfectly.

1) `Component` has been changed to `ItemComponent`. This causes an issue in that the ActionRow.of() method supports collections of ItemComponent. This can be fixed by changing the type of the row lists under the pagination class, such as line 1610

2) Buttons (and `ButtonInteraction` classes )have been moved in the JDA under a category of their own, from "".components.Button to "".components.buttons.Button and "".components.buttons.ButtonInteraction

3) `OnButtonClick` events have been renamed to `ButtonInteractionEvent`. However its methods are the same, so changing imports and fixing types also solves this issue

I'm not sure if this library is still supported, but if it is, doing these changes might be helpful for newcomers. Very helpful library in any case!

-Dennis

### PR DESCRIPTION
**Pull request by SIMULATAN**
This PR adds support for JDA 5 (currently .20, will change in the future), based on @BlitzOffline's work in #39 (tysm for that!), but unfortunately, they didn't update their branch as newer JDA alpha versions with breaking changes came out, so we did that ourselves.

We'll most likely continue to update our fork for JDA 5 versions at least until it's stable.

## Tasks
- [x] bump versions
	- [x] 3.1.0 for .13
	- [X] 3.2.0 for .16
	- [X] 3.3.0 for .20
- [X] updated to Java 17 (+ fix for Jitpack)
- [ ] implemented review changes (I might've, but I'm not sure, I'll check at some point soon)

Closes #46

### CODE REVIEW

**Comment by ygimenez** [comment]
The work both of you are doing is greatly appreciated, although I'm not sure I'll be able to merge those PRs when the stable version gets released since a lot changed in the nightly branch (mostly optimization and reducing unnecessary edits, but the whole structure as well).

Nonetheless, thanks for adding support to alpha JDA.

**Comment by SIMULATAN** [comment]
I depend on the library for my own discord bot, thus I'd be completely fine with also porting these changes.
If you want to, I can make another branch, and try to update yours for alphas, although no guarantee, because I haven't checked out the nightly branch changes yet.

**Comment by ygimenez** [comment]
Don't worry about it, I'm just mentioning that it might not be possible to merge the PRs onto master when stable gets released, I'm totally fine with you guys having your own forks.

File: pom.xml [hunk 1]
```diff
@@ -55,8 +55,8 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>3.8.1</version>
                 <configuration>
-                    <source>9</source>
-                    <target>9</target>
+                    <source>17</source>
```
Comment (ygimenez) [inline pom.xml:58]: Library cannot depend on newer JDK versions since it'll prevent users with older versions from using this library

Comment (SIMULATAN) [reply pom.xml:58]: Isn't this PR obsolete now anyway?

File: src/main/java/com/github/ygimenez/method/Pages.java [hunk 2]
```diff
@@ -9,17 +9,19 @@
 import com.github.ygimenez.type.Emote;
```
Comment (ygimenez) [inline src/main/java/com/github/ygimenez/method/Pages.java:9]: Pattern-matching if clauses not available at lower JDK versions

**Comment by ygimenez** [comment]
Due to diverging structures I cannot merge this PR, but I thank you for maintaining this fork while JDA was still in alpha stage.

**Pull request closed without merging**